pkg/crc/image/image.go

```diff
@@ -0,0 +1,130 @@
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	crcpreset "github.com/code-ready/crc/pkg/crc/preset"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/code-ready/crc/pkg/extract"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/directory"
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/types"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type imageHandler struct {
+	imageURI string
+}
+
+func uri(preset crcpreset.Preset) string {
+	return fmt.Sprintf("//%s/%s:%s", constants.RegistryURI, getImageName(preset), version.GetCRCVersion())
+}
+
+func (img *imageHandler) policyContext() (*signature.PolicyContext, error) {
+	policy := &signature.Policy{Default: []signature.PolicyRequirement{signature.NewPRInsecureAcceptAnything()}}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return nil, fmt.Errorf("error creating security context: %w", err)
+	}
+
+	return policyContext, nil
+}
+
+func (img *imageHandler) getSystemContext() *types.SystemContext {
+	return &types.SystemContext{
+		OSChoice:           runtime.GOOS,
+		ArchitectureChoice: runtime.GOARCH,
+	}
+}
```

Comment:
fwiw, these are the default values
```
        // If not "", overrides the use of platform.GOARCH when choosing an image or verifying architecture match.
        ArchitectureChoice string
        // If not "", overrides the use of platform.GOOS when choosing an image or verifying OS match.
        OSChoice string
```
probably not worth the helper/..., we can just use an empty context.